libdrgn/symbol.c

```diff
@@ -73,3 +73,61 @@ LIBDRGN_PUBLIC bool drgn_symbol_eq(struct drgn_symbol *a, struct drgn_symbol *b)
 		a->size == b->size && a->binding == b->binding &&
 		a->kind == b->kind);
 }
+
+DEFINE_VECTOR_FUNCTIONS(symbolp_vector);
+
+LIBDRGN_PUBLIC struct drgn_error *
```

Comment:
I think it makes sense to switch to the convention. Especially since it would have made explicit the cases below where you caught some leaks in error cases: I would have needed to handle the error and probably realized I needed to free a symbol.

Reply:
Haha, this even revealed another leak in the kallsyms symbol finder.

I think I just assumed that on error, the builder would free the symbol. Which may be convenient in some cases, but isn't a good generic API.